### ISSUE TITLE
[Bug]: Combat Bar (Instant Casting) causes different Problems and Exceptions upon rejoining.

### ISSUE THREAD

**Issue by Colderjacket** [promcteam/proskillapi#254]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you tried to reproduce this without OP?

- [X] I have tested without OP

### Current Behavior

When i put Items into the Combat Bar, it saves the items normally.
But as i rejoin the Server the Items in the Combat bar loses their NBTData.
In some cases it is also causing an Exception, removing all items in the CombatBar and also removing the "Swap Hotbars" item.
The Exception is mostly caused when i store items of the MMOItems Plugin from TeamRequiem in the Combat Hotbar.

### Expected Behavior

Back then in older versions of skillAPI everything regarding that worked fine.

### Steps To Reproduce

Create an Item with custom NBT Data for example:
/give @p minecraft:iron_sword{AttributeModifiers:[{AttributeName:generic.attack_damage,Amount:8,Slot:mainhand,Operation:0,UUID:[I;13243,1439,1767,44564]}]}

Then put that item into the Combat Bar.

When leaving it only saves the Item Type in the uuid yml file.
combatItems: '9;3#t@IRON_SWORD;'

So when i rejoin it loads a standard Iron Sword.
  
The Exception (example with MMOItems plugin)

Take an existing or create a new MMOItem.
Put it into the Combat bar.
Rejoin.
Then the Items in the Combat bar including the Swap Hotbars item disappear.
In Addition an exception is thrown (posted down below).

### Environment

```markdown
- Server Software: Purpur Spigot 1.18.1
- ProSkillAPI Version: 1.1.7.14-ALPHA
- ProMCCore Version: 1.0.3.6-ALPHA
- Plugins: AnimatedScoreboard, AuctionHouse, BeautyQuests, ChestCommands, Citizens, Denizen, DiscoFloor, DiscordSRV, EffectLib, Essentials, EssentialsChat, EssentialsSpawn, F3NPerm, GemsEconomy, HeadDB, ImageOnMap, LibsDisguises, LuckPerms, MCJukebox, MMOItems, MP_dev_GUI, Multiverse-Core, MythicLib, MythicMobs, PhatLoots, PlaceholderAPI, ProMCCore, ProSkillAPI, ProtocolLib, Shopkeepers, Vault,WorldEdit, WorldGuard.

I also tried a test server with proskillapi and promccore only. Nothing changed.
```

### latest.log

The error that occurs when joining depending on the Item in the Combat Bar.

Could not pass event PlayerJoinEvent to ProSkillAPI v1.1.7.14-ALPHA
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at com.sucy.skill.api.util.ItemSerializer.basicDeserialize(ItemSerializer.java:360) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at com.sucy.skill.api.util.ItemSerializer.fromBase64(ItemSerializer.java:215) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at com.sucy.skill.listener.CastCombatListener.init(CastCombatListener.java:92) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at com.sucy.skill.listener.MainListener.lambda$init$0(MainListener.java:121) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.sucy.skill.listener.MainListener.init(MainListener.java:121) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at com.sucy.skill.listener.MainListener.onJoin(MainListener.java:169) ~[proskillapi-1.1.7.14-ALPHA.jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor520.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:git-Purpur-1496]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:630) ~[purpur-api-1.18.1-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.players.PlayerList.postChunkLoadJoin(PlayerList.java:359) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.players.PlayerList.lambda$placeNewPlayer$1(PlayerList.java:301) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:323) ~[?:?]
	at net.minecraft.network.Connection.tick(Connection.java:526) ~[?:?]
	at net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:201) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1664) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:519) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1497) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1285) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at net.minecraft.server.MinecraftServer.lambda$spin$1(MinecraftServer.java:322) ~[purpur-1.18.1.jar:git-Purpur-1496]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

### Anything else?

Just an idea of a fix if the problem is what i think of it is:
Maybe you could add an option to change how the Items are stored in the ymls?
Like some other plugins do, you could maybe store the complete ItemStack instances in a seperate yml file?

**Comment by Travja** [comment]
The error being thrown should be addressed in upcoming builds. Whether this fixes serialization or not is yet to be determined.

**Comment by Travja** [comment]
Is this issue still present?

**Comment by Travja** [comment]
This should be officially resolved with regards to #296. Let me know if any other errors are still present.
